src/protocols/omron/conn.c

```diff
@@ -1499,28 +1501,44 @@ int process_requests(omron_conn_p conn)
             /* if there are still requests after purging all the aborted requests, process them. */
 
             /* how much space do we have to work with. */
-            remaining_space = max_payload_size - (int)sizeof(cip_multi_req_header);
+            remaining_request_space = conn->max_payload_size - (int)sizeof(cip_multi_req_header);
+
+             /* -2 bytes for the msp (multi service packet) number of packets and -4 bytes for the cip response header,
+              * we later subtract 2 bytes for each packet, this is to allow for the INT offset to the packet within the msp 
+              * finally we subtract 10 bytes just to give a comfort blanket as I had seen PLC_TAG_TOO_LARGE issue without it due to too much
+              * data being packed into a single packet */
+            remaining_response_space = conn->max_payload_size - 2 - 4 -10;
```

Comment:
We could just set up a defined constant called "OMRON_OVERHEAD_FUDGE" or something.   It would at least let us know what the numbers were for.